addons/mail/static/tests/core/new_message_separator.test.js

```diff
@@ -251,20 +251,19 @@ test("show new message separator when message is received while chat window is c
         channel_type: "chat",
     });
     onRpcBefore("/mail/data", (args) => {
-        if (args.fetch_params.includes("init_messaging")) {
+        if (JSON.stringify(args.fetch_params).includes("discuss.channel")) {
             asyncStep(`/mail/data - ${JSON.stringify(args)}`);
         }
     });
+    onRpcBefore("/web/dataset/call_kw/ir.http/lazy_session_info", (args) => {
+        asyncStep("init_messaging");
+    });
```

Comment:
If `init_messaging` and `/data` (`mail.thread`/`discuss.channel`) are split it defeats the whole purpose (there are no less RPC) as soon as there is at least one chat window. Initial channel/thread restore from chat hub should also be included in `lazy_session_info` for this change to be fully worth it.